src/rust/engine/logging/src/logger.rs

```diff
@@ -34,6 +34,7 @@ pub struct Logger {
   stderr_log: Mutex<MaybeWriteLogger<Stderr>>,
   show_rust_3rdparty_logs: AtomicBool,
   engine_display_handles: Mutex<HashMap<Uuid, Arc<Mutex<EngineDisplay>>>>,
+  console_display_callback: Mutex<Box<dyn Fn(String) -> () + Send>>,
```

Comment:
Similar to the engine_display, this should probably be a list of callbacks (could probably replace that field, in fact?), for the same reason: we will absolutely want to support multiple clients within the same server within the next few months, and it's already possible to have two concurrent calls to `Scheduler::execute` (although both of them having displays attached is unluckily).

I remember that you used `Fn(String)` here to avoid a cycle between packages, but I think that similar to the `EngineDisplay` class, the new UI code should probably be in a separate crate.

Reply:
The problem I had trying to move this code into a new crate was a dependency on `RootResult`, which is defined in the main engine crate and is used all over the place.